server/service/auth/index.js

```diff
@@ -1,39 +1,46 @@
 const { Router } = require('express');
 
 const {
-    requireUser,
-    respondWithUser,
     createUser,
-    loginUser
+    loginUser,
+    requireUser,
+    respondWithUser
 } = require('./middleware');
 const { checkForDuplicate } = require('./middleware');
 const {
     validateRequestUsernameAndEmail,
     validateRequestBody
 } = require('../../util/requestValidation');
 
-const authRouter = Router();
+const router = Router();
 
-authRouter.get('/me', requireUser, respondWithUser);
+router.get('/me', requireUser, respondWithUser);
 
-authRouter.post('/signup', [
+router.post('/signup', [
     validateRequestBody,
     validateRequestUsernameAndEmail,
     checkForDuplicate,
     createUser,
     respondWithUser
 ]);
 
-authRouter.post('/login', [
+router.get('/signup/usernames/:username/exists', [
+    checkForDuplicate,
```

Comment:
Hooray for existing middleware!

Reply:
Yeah, big time! This required almost NO changes to the backend.